std-bits/table/src/main/java/org/enso/table/data/column/storage/DoubleStorage.java

```diff
@@ -242,6 +243,27 @@ protected double doDouble(
                 return Math.pow(a, b);
               }
             })
+        .add(
+            new UnaryIntegerOp<>(Maps.TRUNCATE) {
+              @Override
+              protected long doOperation(Double a) {
+                return (long) a.doubleValue();
+              }
+            })
```

Comment:
I think we should implement an `UnaryDoubleToIntegerOp`.

I'm not sure how big the impact on performance will be (I can see that with your changes we _already_ have ~50x improvement, which is great), but overall in the Table Java helpers we are trying to avoid boxing wherever possible.

The `UnaryIntegerOp` was made for values that are not primitive types, like `LocalDate`, hence they always are passed as references.

But for primitives, we are boxing the `long` value into a heap allocated `Long` (possibly the JVM can optimize out some of these allocations, but there is some overhead).

So I suggest we implement a separate class for `double` and `long` operations so that the `doOperation` can take primitive `long`/`double` and not a boxed one. This is really annoying that in Java we cannot rely on generics for the primitives and instead have to write a separate implementation for each primitive type - but that's what we have to live with unfortunately.

Reply:
Done.